File: src/main/java/com/dochiri/hexagonal/presentation/auth/response/RegisterUserResponse.java
package com.dochiri.hexagonal.presentation.auth.response;

import com.dochiri.hexagonal.application.user.dto.RegisterUserResult;

public record RegisterUserResponse(
        String id,
        String email
) {
    public static RegisterUserResponse from(RegisterUserResult result) {
        return new RegisterUserResponse(result.id(), result.email());
    }
}

File: src/main/kotlin/com/dochiri/hexagonal/presentation/auth/response/RegisterUserResponse.kt
package com.dochiri.hexagonal.presentation.auth.response

import com.dochiri.hexagonal.application.user.dto.RegisterUserResult

data class RegisterUserResponse(
    val id: String,
    val email: String
) {
    companion object {
        fun from(result: RegisterUserResult): RegisterUserResponse =
            RegisterUserResponse(result.id, result.email)
    }
}
